fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -200,40 +207,70 @@ public DataApiRequest(
 
         DateTimeFormatter dateTimeFormatter = generateDateTimeFormatter(timeZone);
 
-        this.intervals = generateIntervals(intervals, this.granularity, dateTimeFormatter);
-
-        metricDictionary = metricDictionary.getScope(Collections.singletonList(tableName));
+        this.intervals = timeGenerator(
+                () -> generateIntervals(intervals, this.granularity, dateTimeFormatter),
+                "GeneratingIntervals"
+        );
 
         this.filterBuilder = bardConfigResources.getFilterBuilder();
 
+        MetricDictionary metricDictionary = bardConfigResources
+                .getMetricDictionary()
+                .getScope(Collections.singletonList(tableName));
```

Comment:
Because we were originally assigning the `metricDictionary` variable at the very beginning of the method, but didn't do anything with it until we reassigned it to the above value just before actually using it.

Since we were reassigning it, it was no longer effectively final. So I just got rid of the unused initialization.